pagerduty/resource_pagerduty_event_orchestration_integration.go

```diff
@@ -0,0 +1,309 @@
+				if readDestErr != nil {
+					log.Printf("[WARN] Integration '%s' cannot be found on the destination PagerDuty Event Orchestration '%s'. Retrying migration....", id, destinationOrchId)
+					return resource.RetryableError(readDestErr)
+				}
+
+				// retry migration if the integration still exists on the source:
+				if readSrcErr == nil && srcInt != nil {
+					log.Printf("[WARN] Integration '%s' still exists on the source PagerDuty Event Orchestration '%s'. Retrying migration....", id, sourceOrchId)
+					return resource.RetryableError(fmt.Errorf("Integration '%s' still exists on the source PagerDuty Event Orchestration '%s'.", id, sourceOrchId))
+				}
+			}
+			return nil
+		})
+
+		if retryErr != nil {
+			time.Sleep(2 * time.Second)
+			return diag.FromErr(retryErr)
+		}
+	}
+
+	// Update migrated integration if the label property was modified
+	if d.HasChange("label") {
+		oid, payload := getEventOrchestrationIntegrationPayloadData(d)
+
+		retryErr := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+			log.Printf("[INFO] Updating Integration '%s' for PagerDuty Event Orchestration: %s", id, oid)
+
+			if integration, _, err := client.EventOrchestrationIntegrations.UpdateContext(ctx, oid, id, payload); err != nil {
+				if isErrCode(err, 400) {
+					return resource.NonRetryableError(err)
+				}
+				return resource.RetryableError(err)
+			} else if integration != nil {
+				// Try reading an integration after updating the label, retry if the label is not updated:
+				if updInt, _, readErr := client.EventOrchestrationIntegrations.GetContext(ctx, oid, id); readErr == nil && updInt.Label != payload.Label {
+					log.Printf("[WARN] Label for Integration '%s' on PagerDuty Event Orchestration '%s' was not updated. Expected: '%s', actual: '%s'. Retrying update...", id, oid, payload.Label, updInt.Label)
+					return resource.RetryableError(fmt.Errorf("Label for Integration '%s' on PagerDuty Event Orchestration '%s' was not updated.", id, oid))
+				} else {
+					setEventOrchestrationIntegrationProps(d, integration)
+				}
+			}
+			return nil
+		})
+
+		if retryErr != nil {
+			time.Sleep(2 * time.Second)
+			return diag.FromErr(retryErr)
+		}
+	}
+
+	return nil
+}
+
+func resourcePagerDutyEventOrchestrationIntegrationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := d.Id()
+	oid, _ := getEventOrchestrationIntegrationPayloadData(d)
+
+	retryErr := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+		log.Printf("[INFO] Deleting Integration '%s' for PagerDuty Event Orchestration: %s", id, oid)
+		if _, err := client.EventOrchestrationIntegrations.DeleteContext(ctx, oid, id); err != nil {
+			return resource.RetryableError(err)
+		} else {
+			// Try reading an integration after deletion, retry if still found:
```

Comment:
TODO: replace with Fetch function

Reply:
Same here, I left the call to `.GetContext()` instead of using the fetch function because we don't need to set any of the integration properties when we delete an integration resource, we are just retrying the deletion process if the integration still exists.